subiquitycore/tests/test_tui.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2023 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import pathlib
+import tempfile
+import unittest
+from unittest.mock import Mock, patch
+
+from subiquitycore.tui import TuiApplication
+
+
+class TestTuiApplication(unittest.TestCase):
+    def setUp(self):
+        with patch("subiquitycore.tui.Application.__init__", return_value=None):
+            opts = Mock()
+            opts.answers = None
+            opts.dry_run = True
+            self.tui = TuiApplication(opts)
+            # Usually, the below are assigned by Application.__init__()
+            self.tui.opts = opts
+            tmpdir = tempfile.TemporaryDirectory()
```

Comment:
SubiTestCase has a tmp_dir() function which would help simplify this a bit.

Reply:
Thanks! Yes that looks simpler.